src/components/NewCommentForm.tsx

```diff
@@ -1,37 +1,140 @@
-import React from 'react';
+import React, { useState, useContext } from 'react';
+import classNames from 'classnames';
+import { GlobalContext } from './GeneralContext';
+import * as commentsService from '../api/comments';
 
 export const NewCommentForm: React.FC = () => {
+  const {
+    comments,
+    selectedPost,
+    setComments,
+  } = useContext(GlobalContext);
+
+  const [newName, setNewName] = useState('');
+  const [newEmail, setNewEmail] = useState('');
+  const [newMessage, setNewMessage] = useState('');
+
+  const [isFormSending, setIsFormSending] = useState(false);
+
+  const [isNameValid, setIsNameValid] = useState(true);
+  const [isEmailValid, setIsEmailValid] = useState(true);
+  const [isMessageValid, setIsMessageValid] = useState(true);
+
+  const handleSetNewName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewName(event.target.value);
+    setIsNameValid(true);
+  };
+
+  const handleSetNewEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewEmail(event.target.value);
+    setIsEmailValid(true);
+  };
+
+  const handleSetNewMessage
+  = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setNewMessage(event.target.value);
+    setIsMessageValid(true);
+  };
+
+  const formValidate = () => {
+    let isFormValid = true;
+
+    if (newName.trim() === '') {
```

Comment:
```suggestion
    if (!newName.trim()) {
```
it's shorter - empty string is a [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value